contracts/RelayRecipient.sol

```diff
@@ -65,20 +65,29 @@ contract RelayRecipient is RelayRecipientApi {
         return orig_msg_data;
     }
 
-	//Contract must inherit and re-implement this method.
-	// return "0" if the the contract is willing to accept the charges from this sender, for this transaction.
-	// 	any other value is a failure. actual value is for diagnostics only.
-	//  values below 10 are reserved by can_relay
-	// @param relay the relay that attempts to relay this function call. 
-	//			the contract may restrict some encoded functions to specific known relays.
-	// @param from the sender (signer) of this function call. 
-	// @param encoded_function the encoded function call (without any signature). 
-	//			the contract may check the method signature for valid methods
-    function may_relay(address /* relay */, address from, bytes /* encoded_function */) public view returns(uint32) {
-        // Inherited and implemented by the recipient contract.  Returns 0 if it's willing to accept the charges of this sender, for that transaction.
-		// any other value is failure
-        require(msg.sender != from); // Just to prevent a warning
-        return 99;
+    /*
+	 * Contract must inherit and re-implement this method.
+	 *  @return "0" if the the contract is willing to accept the charges from this sender, for this function call.
+	 *  	any other value is a failure. actual value is for diagnostics only.
+	 *   values below 10 are reserved by can_relay
+	 *  @param relay the relay that attempts to relay this function call.
+	 * 			the contract may restrict some encoded functions to specific known relays.
+	 *  @param from the sender (signer) of this function call.
+	 *  @param encoded_function the encoded function call (without any ethereum signature).
+	 * 			the contract may check the method-id for valid methods
+	 */
+    function accept_relayed_call(address relay, address from, bytes encoded_function ) public view returns(uint32);
```

Comment:
I just realized that we only pass the encoded function call, not the entire transaction. So accept_relayed_call can't take all factors into account.  E.g. transaction_fee and gas_price.
It's likely that some dapps will want to restrict collect-calls to a certain price limit.  Time-critical apps will accept the charges, but games, for example, might prefer to become unavailable under extreme network conditions and not incur excessive costs.
So shouldn't we include at least gas_price and transaction_fee in the args here?  Anything else that might get factored into the decision?

Reply:
Just to clarify, the reason to add gas_price is for the view-function-calls.  During the actual transaction, the dapp can look at tx.gasPrice, but it's harder for the relay to know whether a transaction will be compensated this way.  And re transaction_fee, that's required anyway, as it's not part of tx.

Reply:
right. its missing. I just hope we didn't reach the # of args limit...
note, however, that gas-price is problematic to use.

but anyway, its a different commit.